draft-ietf-quic-multipath.md

```diff
@@ -1227,11 +1322,153 @@ PATH_AVAILABLE frame is considered lost, the peer SHOULD resend the frame
+
+Stateless Reset Token:
+A 128-bit value that will be used for a stateless reset when the associated 
+connection ID is used.
+
+The Sequence Number field and Retire Prior To field is allocated 
+for each path independently. The Retire Prior To field indicates which connection IDs 
+should be retired on the corresponding path of Path Identifier.
+
+The Retire Prior To field applies to connection IDs established during 
+connection setup if the Path Identifier is 0 indicating the initial path; see {{consume-retire-cid}}. 
+The value in the Retire Prior To field MUST be less than or equal to the value 
+in the Sequence Number field. Receiving a value in the Retire Prior To field 
+that is greater than that in the Sequence Number field MUST be treated as 
+a connection error of type FRAME_ENCODING_ERROR.
+
+Length, Connection ID, Stateless Reset Token fields have exactly the same
+definition in NEW_CONNECTION_ID frame {{Section 19.15 of QUIC-TRANSPORT}}.
+
+Note that Connection IDs issued in NEW_CONNECTION_ID frames MUST be treated as
+their Path Identifier is 0. Also the retire prior to field of NEW_CONNECTION_ID frames
+just effect the Connection IDs of initial path with path ID 0. This machanism 
+is compatible with {{QUIC-Transport}}.
+
+
+## MP_RETIRE_CONNECTION_ID frames {#mp-retire-conn-id-frame}
+
+An endpoint sends a MP_RETIRE_CONNECTION_ID frame (type=0x15228c0a) instead of 
+RETIRE_CONNECTION_ID frame to indicate that it will no longer use a connection ID 
+that was issued by its peer. This includes the connection ID provided during the handshake. 
+Sending a MP_RETIRE_CONNECTION_ID frame also serves as a request to the peer 
+to send additional connection IDs for future use, unless the path specified 
+by Path Identifier has been abandoned. New connection IDs can be 
+delivered to a peer using the MP_NEW_CONNECTION_ID frame ({{mp-new-conn-id-frame}}).
+
+Retiring a connection ID invalidates the stateless reset token associated with that connection ID.
+
+MP_RETIRE_CONNECTION_ID frames are formatted as shown in {{fig-mp-retire-connection-id-frame-format}}.
+
+~~~
+MP_RETIRE_CONNECTION_ID Frame {
+  Type (i) = 0x15228c0a,
+  Path Identifier (i),
+  Sequence Number (i),
+}
+~~~
+{: #fig-mp-retire-connection-id-frame-format title="MP_RETIRE_CONNECTION_ID Frame Format"}
+
+Path Identifier:
+: A path identifier which is pre allocated when the Connection ID is generated, which
+means the current Connection ID can only be used on the corresponding path.
+
+Sequence Number:
+The sequence number assigned to the connection ID by the sender on the path 
+specified by Path Identifier, encoded as a variable-length integer. 
+
+
+## MAX_PATHS frames {#max-paths-frame}
+
+A MAX_PATHS frame (type=0x15228c0b) informs the peer of the cumulative number of paths 
+it is permitted to open. 
+
+MAX_PATHS frames are formatted as shown in {{fig-max-paths-frame-format}}.
+
+~~~
+MAX_PATHS Frame {
+  Type (i) = 0x15228c0b,
+  Maximum Paths (i),
```

Comment:
I think this should be "Maximum Path Id"

Reply:
Note that `Maximum Paths` is symmetric to `Maximum Streams` in `MAX_STREAMS` frames, so keeping the field name might actually be clearer.

Reply:
Leave it to max paths -- as Michael says, this is the same pattern as MAX_STREAMS.

Reply:
No strong opinion on the name, keeping things consistent seems fine. However, I had to read this thread to understand what was actually meant. It would probably help to add a description below the frame that describes that this is the maximum path ID.